config/keys.go

```diff
@@ -29,29 +36,151 @@ type KeySuite struct {
 	*ExternalKey
 }
 
-// Is checks whether the given name is equal with the Name variable
-func (k KeySuite) Is(name string) bool {
-	return k.Name == name
-}
+var errorKeyNameEmpty = errors.New("key name cannot be empty")
+var errKeyNotFound = errors.New("signing key not found")
 
-// SigningKeys reflects the signingkeys.json file.
+// SigningKeys reflects the signings.json file.
 type SigningKeys struct {
-	Default string     `json:"default"`
+	Default *string    `json:"default"`
 	Keys    []KeySuite `json:"keys"`
 }
 
+// NewSigningKeys creates a new signingkeys config file
+func NewSigningKeys() *SigningKeys {
+	return &SigningKeys{Keys: []KeySuite{}}
+}
+
+// Add adds new signing key
+func (s *SigningKeys) Add(name, keyPath, certPath string, markDefault bool) error {
+	if name == "" {
+		return errorKeyNameEmpty
+	}
+	_, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		return err
+	}
+
+	ks := KeySuite{
+		Name: name,
+		X509KeyPair: &X509KeyPair{
+			KeyPath:         keyPath,
+			CertificatePath: certPath,
+		},
+	}
+	return s.addCore(ks, markDefault)
+}
+
+// AddPlugin adds new plugin based signing key
+func (s *SigningKeys) AddPlugin(ctx context.Context, keyName, id, pluginName string, pluginConfig map[string]string, markDefault bool) error {
+	logger := log.GetLogger(ctx)
+	logger.Debugf("Adding key with name %v and plugin name %v", keyName, pluginName)
+
+	if keyName == "" {
+		return errorKeyNameEmpty
+	}
+
+	if id == "" {
+		return errors.New("missing key id")
+	}
+
+	if pluginName == "" {
+		return errors.New("plugin name cannot be empty")
+	}
+
+	mgr := plugin.NewCLIManager(dir.PluginFS())
+	_, err := mgr.Get(ctx, pluginName)
+	if err != nil {
+		return err
+	}
+
+	ks := KeySuite{
+		Name: keyName,
+		ExternalKey: &ExternalKey{
+			ID:           id,
+			PluginName:   pluginName,
+			PluginConfig: pluginConfig,
+		},
+	}
+
+	err =  s.addCore(ks, markDefault)
+	logger.Debugf("Added key with name %s - {%+v}", keyName, ks)
+	return err
```

Comment:
When err is not nil, the log still claims "Added key with name...". Should we do something like:
```suggestion
	if err :=  s.addCore(ks, markDefault); err != nil {
	    logger.Errorf(xxx)
	    return err
	}
	logger.Debugf("Added key with name %s - {%+v}", keyName, ks)
	return nil
```